Just tested it, the packet is not fully read:

```
[2019-04-29T16:04:46+0200] [FINE] io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Packet "ServerUpdateLightPacket" not fully read.
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:459)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
        at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
        at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:297)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:413)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
        at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
        at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.oio.AbstractOioByteChannel.doRead(AbstractOioByteChannel.java:168)
        at io.netty.channel.oio.AbstractOioChannel$1.run(AbstractOioChannel.java:37)
        at io.netty.channel.ThreadPerChannelEventLoop.run(ThreadPerChannelEventLoop.java:67)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Packet "ServerUpdateLightPacket" not fully read.
        at com.github.steveice10.packetlib.tcp.TcpPacketCodec.decode(TcpPacketCodec.java:44)
        at io.netty.handler.codec.ByteToMessageCodec$1.decode(ByteToMessageCodec.java:42)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
        ... 38 more

```

--- a/src/main/java/com/github/steveice10/mc/protocol/packet/ingame/server/world/ServerUpdateLightPacket.java
+++ b/src/main/java/com/github/steveice10/mc/protocol/packet/ingame/server/world/ServerUpdateLightPacket.java
@@ -62,6 +62,9 @@ public class ServerUpdateLightPacket extends MinecraftPacket {
         this.skyLight = new ArrayList<>(18);
         for (int i = 0; i < 18; i++) {
             if ((skyLightMask & 1 << i) != 0) {
+                if (in.readVarInt() != 2048) {
+                    throw new IOException("Expected sky light byte array to be of length 2048");
+                }
                 this.skyLight.add(new NibbleArray3d(in, 2048)); // 2048 bytes read = 4096 entries
             } else if ((emptySkyLightMask & 1 << i) != 0) {
                 this.skyLight.add(new NibbleArray3d(4096));
@@ -73,6 +76,9 @@ public class ServerUpdateLightPacket extends MinecraftPacket {
         this.blockLight = new ArrayList<>(18);
         for (int i = 0; i < 18; i++) {
             if ((blockLightMask & 1 << i) != 0) {
+                if (in.readVarInt() != 2048) {
+                    throw new IOException("Expected block light byte array to be of length 2048");
+                }
                 this.blockLight.add(new NibbleArray3d(in, 2048)); // 2048 bytes read = 4096 entries
             } else if ((emptyBlockLightMask & 1 << i) != 0) {
                 this.blockLight.add(new NibbleArray3d(4096));
@@ -118,12 +124,14 @@ public class ServerUpdateLightPacket extends MinecraftPacket {
 
         for (int i = 0; i < 18; i++) {
             if ((skyLightMask & 1 << i) != 0) {
+                out.writeVarInt(2048); // dunno why Minecraft feels the need to send these
                 this.skyLight.get(i).write(out);
             }
         }
 
         for (int i = 0; i < 18; i++) {
             if ((blockLightMask & 1 << i) != 0) {
+                out.writeVarInt(2048); // dunno why Minecraft feels the need to send these
                 this.blockLight.get(i).write(out);
             }
         }

